CMakeLists.txt

```diff
@@ -221,6 +227,9 @@ if (CMAKE_BUILD_TYPE STREQUAL "Debug")
 	add_definitions(-DDEBUG_ENABLED)
 	message(STATUS "Using compilation flags: ${CMAKE_CXX_FLAGS_DEBUG}")
 elseif (CMAKE_BUILD_TYPE STREQUAL "Release")
+	# By deafult Release build uses -O3 which usually is not the best idea.
+	string(REPLACE "-O3" "-O2" CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE}")
```

Comment:
What does -O3 break?

Reply:
Technically nothing I'm aware of (at least in `dg`) but `-O3` can generate bigger binaries and thus potentially slower code and it can also make the compilation slower. If you insist I can definitely drop this change as the new default build type `RelWithDebInfo` uses `-O2` implicitly.

Unfortunatelly, my laptop stopped working so I will not able to make the proposed changes (or new PR's) until I get a replacement one. Hopefully, it will not take more than a few weeks.

Reply:
Yes, drop this change, please. Let cmake choice the best flags. The user can override them if something goes wrong.